AltServer/AppDelegate.swift

```diff
@@ -479,13 +433,47 @@ private extension AppDelegate
 
 extension AppDelegate: NSMenuDelegate
 {
+    
+    func menuWillOpen(_ menu: NSMenu)
+    {
+        if menu == self.appMenu
+        {
+            self.connectedDevices = ALTDeviceManager.shared.connectedDevices
+            
+            self.launchAtLoginMenuItem.target = self
+            self.launchAtLoginMenuItem.action = #selector(AppDelegate.toggleLaunchAtLogin(_:))
+            self.launchAtLoginMenuItem.state = LaunchAtLogin.isEnabled ? .on : .off
+            
+            if self.isMailPluginInstalled
+            {
+                self.installMailPluginMenuItem.title = NSLocalizedString("Uninstall Mail Plug-in", comment: "")
+            }
+            else
+            {
+                self.installMailPluginMenuItem.title = NSLocalizedString("Install Mail Plug-in", comment: "")
+            }
+            self.installMailPluginMenuItem.target = self
+            self.installMailPluginMenuItem.action = #selector(AppDelegate.handleInstallMailPluginMenuItem(_:))
+        }
+    }
+
     func numberOfItems(in menu: NSMenu) -> Int
     {
+        if menu != self.connectedDevicesMenu
+        {
+            return -1
+        }
+        
         return self.connectedDevices.isEmpty ? 1 : self.connectedDevices.count
     }
     
     func menu(_ menu: NSMenu, update item: NSMenuItem, at index: Int, shouldCancel: Bool) -> Bool
     {
+        if menu != self.connectedDevicesMenu
+        {
+            return false
+        }
```

Comment:
Another `if` -> `guard` 

```suggestion
        guard menu == self.connectedDevicesMenu else { return }
```